### ISSUE TITLE
Correct handling of latex special chars

### ISSUE THREAD

**Issue by oberien** [oberien/pundoc#53]
Latex has [10 special chars](http://www.emerson.emory.edu/services/latex/latex_155.html): `# $ % & ~ _ ^ \ { }` Currently (after #47) we translate them as text symbol (instead of as control character), whenever they are encountered (except `\`, which we're checking for a possible latex command and only escape if it is definitely not a command). We may want to handle them context-awarely to make inline latex work as expected.

Another idea is to have a special way of defining inline latex, which is translated differently, like
````markdown
```inlinelatex
\whatevercommandyouwanttouse
```
````

That way we can assume that no inline-latex will be present and just translate everything. If we go this way we could just drop special blocks in all other backends and warn the user. The user can define different block for each backend, in wihch case we don't output a warning:

````
```inlinelatex
\whatever
```
```inlinehtml
<whatever />
```
````
(Ok, bad example, because inline-html should already be supported according to CommonMark anyways. We could think about disallowing it anyways and forcing this method onto our users instead.)

**Comment by 197g** [comment]
One problem with code blocks is that I typically may want to *mix* latex content with heradoc to get the best of both worlds. For example to open some custom `minipage` environment. Surrounding everything with triple quoted code blocks looks extremely clunky. Maybe we can leverage inline code blocks for a similar effect:

```
`\begin{minipage}{0.5\line}`
**Here** I would still want the full improvements from heradoc.
`\end{minipage}`
```

This could be complementary to the full code blocks! Basically establishing again that inline code blocks should have the same power as full ones, though it would require some special syntax considerations as it is less obvious than `$ math^2`

**Comment by 197g** [comment]
Related, I think `[include la.tex]` (or `![](la.tex)`) should eventually behave the same as an `inlinelatex` code block as well (I'm fine with doing these in steps). That includes potentially disallowing it without a declared latex feature.

**Comment by oberien** [comment]
The idea is that you are free to mix latex blocks / environments with heradoc (and that's what I'm using it for):
````
```inlinelatex
\begin{landscape}
```
text foo bar
```inlinelatex
\end{landscape}
```
````

I do agree that it would be nice to have some sort of inline-syntax for latex similar to inline math. I don't know which character to promote to a metachar, but my first though would be a backslash instead of a dollar (like math) followed by a space: `\ \foo`.

**Comment by 197g** [comment]
Agree with the choice of metachar to start the inline block, thankfully commonmark [disables escapes within code blocks](https://spec.commonmark.org/0.28/#example-294). I'm only uncertain how to build an escape hatch for this, as well as `$ ` (or if it is necessary).

**Comment by oberien** [comment]
Reopening because this is only partially fixed. Only blocks have been implemented in #56 and not inline-code latex.
